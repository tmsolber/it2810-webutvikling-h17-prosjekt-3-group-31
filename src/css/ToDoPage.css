.btn-container {
    position: relative;
}

.todoTextFalse{
    font-size: 20px;
}

.todoTextTrue{
    font-size: 20px;
    text-decoration: line-through;
    text-decoration-color: dimgrey;
}

.todo-segment {
    padding: 50px 0;
}

#hello {
    cursor: pointer;
    color: rgba(15, 26, 63, 0.68);
    font-size: 25px;
}