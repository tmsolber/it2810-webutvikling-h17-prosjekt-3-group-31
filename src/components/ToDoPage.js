import React, {Component} from 'react';
import ToDoForm from './ToDoForm';
import AddToDo from "./AddToDo";
import '../css/ToDoPage.css';
import {Segment, Checkbox, Grid, Icon} from 'semantic-ui-react'
import BackgroundImage from "../containers/BackgroundImage";

class ToDoPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showModal: false,
            value: (localStorage.getItem('notes') && JSON.parse(localStorage.getItem('notes')))||[],
            check: false
        };
        this.handleModal = this.handleModal.bind(this);
        this.handleInputFromToDo = this.handleInputFromToDo.bind(this);
        this.handleCheck = this.handleCheck.bind(this);
        this.handleDelete = this.handleDelete.bind(this);
    }

    handleModal() {
        this.setState(prevState => ({
            showModal: !prevState.showModal
        }));
    }

    handleInputFromToDo(val){
        this.state.value.push({
            text: val,
            key: Date.now()
        });
        this.setState({
            value: this.state.value
        });
        console.log(this.state.value)
    }

    // Delete the todoitem given by the key, and save the new value of the list.
    handleDelete(key){
        const todos = this.state.value.filter(function(val){
            return (val.key !== key);
        });

        this.setState({
            value: todos
        });
    }

    componentDidUpdate(){
        localStorage.setItem('notes', JSON.stringify(this.state.value));
    }

    handleCheck(){
        if(this.state.check === false){
            this.setState({
                check: true
            });
        }else{
            this.setState({
                check: false
            });
        }
    }

    render() {
        const className = this.state.check ? "todoTextTrue" : "todoTextFalse";
        const listItems = this.state.value.map((v) =>
            <Segment className="TodoItem" key={v.key}>
                <Grid columns={"equal"}>
                    <Grid.Column width={1}>
                        <Checkbox className="checkbox" onClick={(e)=>this.handleCheck()}/>
                    </Grid.Column>
                    <Grid.Column width={13} stretched className={className} name={"todoText"} >
                        {v.text}
                    </Grid.Column>
                    <Grid.Column width={2}>
                        <Icon id="remove-icon" name="remove" onClick={(e) => this.handleDelete(v.key)}/>
                    </Grid.Column>
                </Grid>
            </Segment>
        );

        return (
            <div>
                <BackgroundImage/>
                <div className="btn-container">
                    <AddToDo show={this.handleModal}/>
                </div>
                <Grid className="App">
                    <Grid.Column width={5} textAlign="centered">
                        <Grid.Row>
                            <div className="todo-segment">
                                <Segment.Group>
                                    {listItems}
                                </Segment.Group>
                            </div>
                            <ToDoForm show={this.state.showModal} onClose={this.handleModal}
                                      onUpdate={this.handleInputFromToDo} values={this.state.value}/>
                        </Grid.Row>
                    </Grid.Column>
                </Grid>
            </div>
        );

    }
}

export default ToDoPage;