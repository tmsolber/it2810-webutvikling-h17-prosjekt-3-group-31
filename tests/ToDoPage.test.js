import './globalJSDOM';
import ToDoPage from '../src/components/ToDoPage';
import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock();
configure({ adapter: new Adapter() });

test('test that page returns content without props', () => {
  const component = shallow(<ToDoPage />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('test that page is rendered without values', () => {
  const component = mount(<ToDoPage />);
  const tree = toJson(component);
  expect(component.find('GridColumn.todoTextFalse').length).toBe(0);
  expect(tree).toMatchSnapshot();
});

test('test that values are rendered', () => {
  const values = [
    { text: 'asd', key: 1508355111614, cName: 'todoTextFalse', checked: null },
    { text: 'dsa', key: 1508355113035, cName: 'todoTextFalse', checked: null }
  ];
  const component = mount(<ToDoPage />);
  component.setState({ value: values });
  const tree = toJson(component);
  expect(component.find('GridColumn.todoTextFalse').length).toBe(2);
  expect(tree).toMatchSnapshot();
});
