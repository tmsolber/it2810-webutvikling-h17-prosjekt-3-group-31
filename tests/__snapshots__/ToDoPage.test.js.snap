// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that page is rendered without values 1`] = `
<ToDoPage>
  <div>
    <BackgroundImage />
    <div
      className="btn-container"
    >
      <AddToDo
        show={[Function]}
      >
        <div
          className="add-btn"
        >
          <Popup
            content="Click to add todo"
            on="hover"
            position="top left"
            trigger={
              <img
                alt="add-to-do"
                className="plus"
                onClick={[Function]}
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            }
          >
            <Portal
              closeOnDocumentClick={true}
              closeOnEscape={true}
              closeOnTriggerMouseLeave={true}
              eventPool="default"
              mouseEnterDelay={50}
              mouseLeaveDelay={70}
              onClose={[Function]}
              onMount={[Function]}
              onOpen={[Function]}
              onUnmount={[Function]}
              openOnTriggerClick={true}
              openOnTriggerMouseEnter={true}
              trigger={
                <img
                  alt="add-to-do"
                  className="plus"
                  onClick={[Function]}
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              }
            >
              <img
                alt="add-to-do"
                className="plus"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </Portal>
          </Popup>
        </div>
      </AddToDo>
    </div>
    <Grid
      className="App"
    >
      <div
        className="ui grid App"
      >
        <GridColumn
          textAlign="centered"
          width={5}
        >
          <div
            className="centered aligned five wide column"
          >
            <GridRow>
              <div
                className="row"
              >
                <div
                  className="todo-segment"
                >
                  <SegmentGroup>
                    <div
                      className="ui segments"
                    />
                  </SegmentGroup>
                </div>
                <ToDoForm
                  onClose={[Function]}
                  onUpdate={[Function]}
                  show={false}
                  values={Array []}
                />
              </div>
            </GridRow>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</ToDoPage>
`;

exports[`test that page returns content without props 1`] = `
<div>
  <BackgroundImage />
  <div
    className="btn-container"
  >
    <AddToDo
      show={[Function]}
    />
  </div>
  <Grid
    className="App"
  >
    <GridColumn
      textAlign="centered"
      width={5}
    >
      <GridRow>
        <div
          className="todo-segment"
        >
          <SegmentGroup />
        </div>
        <ToDoForm
          onClose={[Function]}
          onUpdate={[Function]}
          show={false}
          values={Array []}
        />
      </GridRow>
    </GridColumn>
  </Grid>
</div>
`;

exports[`test that values are rendered 1`] = `
<ToDoPage>
  <div>
    <BackgroundImage />
    <div
      className="btn-container"
    >
      <AddToDo
        show={[Function]}
      >
        <div
          className="add-btn"
        >
          <Popup
            content="Click to add todo"
            on="hover"
            position="top left"
            trigger={
              <img
                alt="add-to-do"
                className="plus"
                onClick={[Function]}
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            }
          >
            <Portal
              closeOnDocumentClick={true}
              closeOnEscape={true}
              closeOnTriggerMouseLeave={true}
              eventPool="default"
              mouseEnterDelay={50}
              mouseLeaveDelay={70}
              onClose={[Function]}
              onMount={[Function]}
              onOpen={[Function]}
              onUnmount={[Function]}
              openOnTriggerClick={true}
              openOnTriggerMouseEnter={true}
              trigger={
                <img
                  alt="add-to-do"
                  className="plus"
                  onClick={[Function]}
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              }
            >
              <img
                alt="add-to-do"
                className="plus"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </Portal>
          </Popup>
        </div>
      </AddToDo>
    </div>
    <Grid
      className="App"
    >
      <div
        className="ui grid App"
      >
        <GridColumn
          textAlign="centered"
          width={5}
        >
          <div
            className="centered aligned five wide column"
          >
            <GridRow>
              <div
                className="row"
              >
                <div
                  className="todo-segment"
                >
                  <SegmentGroup>
                    <div
                      className="ui segments"
                    >
                      <Segment
                        className="TodoItem"
                        key="1508355111614"
                      >
                        <div
                          className="ui segment TodoItem"
                        >
                          <Grid
                            columns="equal"
                          >
                            <div
                              className="ui equal width grid"
                            >
                              <GridColumn
                                width={1}
                              >
                                <div
                                  className="one wide column"
                                >
                                  <Checkbox
                                    checked={false}
                                    onClick={[Function]}
                                    type="checkbox"
                                  >
                                    <div
                                      className="ui fitted checkbox"
                                      onChange={[Function]}
                                      onClick={[Function]}
                                      onMouseDown={[Function]}
                                    >
                                      <input
                                        checked={false}
                                        className="hidden"
                                        name={undefined}
                                        readOnly={true}
                                        tabIndex={0}
                                        type="checkbox"
                                        value={undefined}
                                      />
                                      <label />
                                    </div>
                                  </Checkbox>
                                </div>
                              </GridColumn>
                              <GridColumn
                                className="todoTextFalse"
                                stretched={true}
                                width={13}
                              >
                                <div
                                  className="stretched thirteen wide column todoTextFalse"
                                >
                                  asd
                                </div>
                              </GridColumn>
                              <GridColumn
                                width={2}
                              >
                                <div
                                  className="two wide column"
                                >
                                  <Icon
                                    as="i"
                                    id="remove-icon"
                                    name="remove"
                                    onClick={[Function]}
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="remove icon"
                                      id="remove-icon"
                                      onClick={[Function]}
                                    />
                                  </Icon>
                                </div>
                              </GridColumn>
                            </div>
                          </Grid>
                        </div>
                      </Segment>
                      <Segment
                        className="TodoItem"
                        key="1508355113035"
                      >
                        <div
                          className="ui segment TodoItem"
                        >
                          <Grid
                            columns="equal"
                          >
                            <div
                              className="ui equal width grid"
                            >
                              <GridColumn
                                width={1}
                              >
                                <div
                                  className="one wide column"
                                >
                                  <Checkbox
                                    checked={false}
                                    onClick={[Function]}
                                    type="checkbox"
                                  >
                                    <div
                                      className="ui fitted checkbox"
                                      onChange={[Function]}
                                      onClick={[Function]}
                                      onMouseDown={[Function]}
                                    >
                                      <input
                                        checked={false}
                                        className="hidden"
                                        name={undefined}
                                        readOnly={true}
                                        tabIndex={0}
                                        type="checkbox"
                                        value={undefined}
                                      />
                                      <label />
                                    </div>
                                  </Checkbox>
                                </div>
                              </GridColumn>
                              <GridColumn
                                className="todoTextFalse"
                                stretched={true}
                                width={13}
                              >
                                <div
                                  className="stretched thirteen wide column todoTextFalse"
                                >
                                  dsa
                                </div>
                              </GridColumn>
                              <GridColumn
                                width={2}
                              >
                                <div
                                  className="two wide column"
                                >
                                  <Icon
                                    as="i"
                                    id="remove-icon"
                                    name="remove"
                                    onClick={[Function]}
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="remove icon"
                                      id="remove-icon"
                                      onClick={[Function]}
                                    />
                                  </Icon>
                                </div>
                              </GridColumn>
                            </div>
                          </Grid>
                        </div>
                      </Segment>
                    </div>
                  </SegmentGroup>
                </div>
                <ToDoForm
                  onClose={[Function]}
                  onUpdate={[Function]}
                  show={false}
                  values={
                    Array [
                      Object {
                        "cName": "todoTextFalse",
                        "checked": null,
                        "key": 1508355111614,
                        "text": "asd",
                      },
                      Object {
                        "cName": "todoTextFalse",
                        "checked": null,
                        "key": 1508355113035,
                        "text": "dsa",
                      },
                    ]
                  }
                />
              </div>
            </GridRow>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</ToDoPage>
`;
